frontend/src/routes/Governance/common/util.tsx

```diff
@@ -650,18 +651,81 @@ export function hasInformOnlyPolicies(items: Array<PolicyTableItem>): boolean {
   return true
 }
 
+// type guard
+function isPolicy(object: Policy | DiscoverdPolicyItem): object is Policy {
+  return object && (object as Policy).metadata !== undefined
+}
+
 export function getPolicySource(
-  policy: Policy,
+  policy: Policy | DiscoverdPolicyItem,
   helmReleases: HelmRelease[],
   channels: Channel[],
   subscriptions: Subscription[],
   t: TFunction<string, undefined>
 ): string | JSX.Element {
-  const isExternal = resolveExternalStatus(policy)
+  if (!policy) {
+    // Prevent error in detail page
+    return <></>
+  }
+
+  const isExternal = policy && isPolicy(policy) ? resolveExternalStatus(policy) : policy._isExternal || false
   let source: string | JSX.Element = t('Local')
   if (isExternal) {
-    const policySource = resolveSource(policy, helmReleases, channels, subscriptions)
+    const policySource = isPolicy(policy)
+      ? resolveSource(policy.metadata.annotations ?? {}, helmReleases, channels, subscriptions)
+      : resolveSource(policy.annotations ?? {}, helmReleases, channels, subscriptions)
     source = policySource ? getSource(policySource, isExternal, t) : t('Managed externally')
   }
   return source
 }
+
+export function getEngineString(kind: string): string {
+  switch (kind) {
+    case 'ConfigurationPolicy':
+    case 'CertificatePolicy':
+    case 'OperatorPolicy':
+      return 'Open Cluster Management'
+    default:
+      return 'Unknown'
+  }
+}
+
+export function getEngineWithSvg(kind: string): JSX.Element {
+  switch (kind) {
+    case 'ConfigurationPolicy':
+    case 'CertificatePolicy':
+    case 'OperatorPolicy':
+      return (
+        <div style={{ display: 'flex', alignItems: 'center', gap: 8 }}>
+          <img
+            src="https://open-cluster-management.io/ocm.svg"
```

Comment:
Could you please include the SVG in the repo? I don't think we want to be linking to external images.

Reply:
I tried.. I got from ocm webside and saved. it broke the image... Is there anyone who has that svg?

Reply:
![ocm](https://github.com/user-attachments/assets/76a451d8-9542-4c69-8aa4-6b39bcf6d6a9)